src/nvc_container.c

```diff
@@ -236,6 +236,26 @@ nvc_container_new(struct nvc_context *ctx, const struct nvc_container_config *cf
                 error_setx(&ctx->err, "invalid mode of operation");
                 return (NULL);
         }
+        if (flags & OPT_CUDA_COMPAT_MODE_DISABLED) {
+                /*
+                 * If the OPT_CUDA_COMPAT_MODE_DISABLED flag is specified, we
+                 * explicitly ignore other OP_CUDA_COMPAT_MODE_* flags.
+                 */
+                flags &= ~(OPT_CUDA_COMPAT_MODE_MOUNT | OPT_CUDA_COMPAT_MODE_LDCONFIG);
+        } else {
+                if (!(flags & (OPT_CUDA_COMPAT_MODE_LDCONFIG | OPT_CUDA_COMPAT_MODE_MOUNT))) {
+                        /*
+                         * If no OPT_CUDA_COMPAT_MODE_* flags are specified,
+                         * default to OPT_CUDA_COMPAT_MODE_MOUNT to maintain
+                         * backward compatibility.
+                         */
+                        flags &= OPT_CUDA_COMPAT_MODE_MOUNT;
+                }
+        }
+        if ((flags & OPT_CUDA_COMPAT_MODE_MOUNT) && (flags & OPT_CUDA_COMPAT_MODE_LDCONFIG)) {
+                error_setx(&ctx->err, "only one cuda-compat-mode can be specified at a time");
+                return (NULL);
+        }
```

Comment:
Can we extract this into a function / simplify it to modify the flags inline / return an error as necessary.